SCClassLibrary/Common/Core/Condition.sc

```diff
@@ -1,42 +1,121 @@
-Condition {
-	var <>test, waitingThreads;
+// Basic synchronization object
+Sync {
+	var waitingThreads;
 
-	*new { arg test=false;
-		^super.newCopyArgs(test, Array(8))
-	}
-	wait {
-		if (test.value.not, {
-			waitingThreads = waitingThreads.add(thisThread.threadPlayer);
-			\hang.yield;
-		});
+	*new {
+		^super.newCopyArgs(Array(8));
 	}
-	hang { arg value = \hang;
-		// ignore the test, just wait
-		waitingThreads = waitingThreads.add(thisThread.threadPlayer);
+
+	hang { |value = \hang|
+		this.prAddThread(thisThread);
 		value.yield;
 	}
 
-	signal {
-		var tempWaitingThreads, time;
-		if (test.value, {
-			time = thisThread.seconds;
-			tempWaitingThreads = waitingThreads;
-			waitingThreads = nil;
-			tempWaitingThreads.do({ arg thread;
-				thread.clock.sched(0, thread);
-			});
-		});
+	wait {
+		this.hang();
```

Comment:
`this.hang()`

Reply:
I usually leave empty `()` for clarity in cases like this, to emphasize that a method is being called and something is being _done_ (in this case, the something is very critical). Thoughts?

Reply:
I wouldn't have associated any action with empty parentheses. To me,  empty parentheses just make explicit that no arguments are passed.

Strictly speaking, side effects are implementation details. They could be overridden in a subclass. Also, marking side effects with parentheses would fail to communicate as soon as you have to pass arguments.

Reply:
`()` looks like a habit from C, seems foreign in this context. In the main class library, I'd avoid individual style tics, unless there's a concrete reason not to.

`aCondition.hang;` is an established idiom and I think it doesn't require special emphasis.

`()` is common in SC as a shortcut for `.new` but not for much else.

Reply:
We don't have a point in the style guide about this (yet), seems not worth it to bikeshed here